src/containers/EvmCrossChainRecoveryCoin/EvmCrossChainRecoveryCoin.tsx

```diff
@@ -0,0 +1,310 @@
+  }
+  return false;
+}
+
+function getHotWalletCoinForm() {
+  const { env, coin } = useParams<'env' | 'coin'>();
+  const bitGoEnvironment = safeEnv(env);
+  const [, setAlert] = useAlertBanner();
+
+  const navigate = useNavigate();
+
+  switch (coin) {
+    case 'polygon':
+    case 'tpolygon':
+      return (
+        <PolygonFormHotWallet
+          key={coin}
+          onSubmit={async (values, { setSubmitting }) => {
+            setAlert(undefined);
+            setSubmitting(true);
+            try {
+              await window.commands.setBitGoEnvironment(
+                bitGoEnvironment,
+                coin,
+              );
+              const chainData = await window.queries.getChain(coin);
+              const recoverData = await window.commands.recover(coin, {
+                ...values,
+                eip1559: {
+                  maxFeePerGas: toWei(values.maxFeePerGas),
+                  maxPriorityFeePerGas: toWei(values.maxPriorityFeePerGas),
+                },
+                bitgoKey: '',
+                backupKey: '',
+                ignoreAddressTypes: [],
+              });
+              assert(
+                isRecoveryTransaction(recoverData),
+                'Half-signed recovery transaction not detected.'
+              );
+
+              const showSaveDialogData = await window.commands.showSaveDialog({
+                filters: [
+                  {
+                    name: 'Custom File Type',
+                    extensions: ['json'],
+                  },
+                ],
+                defaultPath: `~/${chainData}-half-signed-evmBasedRecovery-${Date.now()}.json`,
+              });
+
+              if (!showSaveDialogData.filePath) {
+                throw new Error('No file path selected');
+              }
+
+              await window.commands.writeFile(
+                showSaveDialogData.filePath,
+                JSON.stringify(
+                  recoverData,
+                  null,
+                  2
+                ),
+                { encoding: 'utf-8' }
+              );
+
+              navigate(
+                `/${bitGoEnvironment}/evm-cross-chain-recovery/${coin}/success`
+              );
+            } catch (err) {
+              if (err instanceof Error) {
+                setAlert(err.message);
+              } else {
+                console.error(err);
+              }
+              setSubmitting(false);
+            }
+          }}
+        />
+      );
+
+    default:
+      throw new Error(`Unsupported coin: ${String(coin)}`);
+  }
+}
+
+function getColdWalletCoinForm() {
+  const { env, coin } = useParams<'env' | 'coin'>();
+  const bitGoEnvironment = safeEnv(env);
+  const [, setAlert] = useAlertBanner();
+
+  const navigate = useNavigate();
+
+  switch (coin) {
+    case 'polygon':
+    case 'tpolygon':
+      return (
+        <PolygonFormColdWallet
+          key={coin}
+          onSubmit={async (values, { setSubmitting }) => {
```

Comment:
please create a named function to handle onSubmit, I think it can be shared between components